internal/pkg/cli/storage_init.go

```diff
@@ -420,6 +485,47 @@ func (o *initStorageOpts) askDynamoLSIConfig() error {
 	}
 }
 
+func (o *initStorageOpts) askAuroraEngineType() error {
+	if o.engine != "" {
+		return nil
+	}
+
+	engine, err := o.prompt.SelectOne(storageInitRDSDBEnginePrompt,
+		storageInitRDSDBEngineHelp,
+		engineTypes,
+		prompt.WithFinalMessage("Database engine:"))
+	if err != nil {
+		return fmt.Errorf("select database engine: %w", err)
+	}
+	o.engine = engine
+	return nil
+}
+
+func (o *initStorageOpts) askAuroraInitialDBName() error {
+	var validator func(interface{}) error
+	switch o.engine {
+	case engineTypeMySQL:
+		validator = validateMySQLDBName
+	case engineTypePostgreSQL:
+		validator = validatePostgreSQLDBName
+	}
+
+	if o.initialDBName != "" {
+		// The flag input is validated here because it needs engine type to determine which validator to use.
+		return validator(o.initialDBName)
+	}
+
+	dbName, err := o.prompt.Get(storageInitRDSInitialDBNamePrompt,
```

Comment:
Maybe we could make use of the selector package here!

Reply:
Ohh do you mean creating a method called `askInitalDBName` in `selector` and then call it here?

Reply:
Yup, and also for selecting engine type.

Reply:
Ahh yess I'm wondering when we do we usually prefer wrapping the prompting in `selector` and when not?